### PR TITLE
feat(models): add optional table title

### PR DESCRIPTION
**Pull request by matejchalk**
Motivation: possible to specify what data the table contains (e.g. _Most frequent errors_).

### CODE REVIEW

File: packages/utils/src/lib/reports/__snapshots__/generate-md-report.unit.test.ts.snap [hunk 1]
```diff
@@ -106,8 +104,6 @@ exports[`generateMdReport > should render complete md report 1`] = `
 <details>
 <summary>🟨 <b>2,7 s</b> (score: 67)</summary>
 
-#### Additional Information
```
Comment (Tlacenka) [inline packages/utils/src/lib/reports/__snapshots__/generate-md-report.unit.test.ts.snap:109]: 🎨 UX consistency: For the sake of consistency of the MD hierarchy, I think we should consider a fallback default value.

Comment (matejchalk) [reply packages/utils/src/lib/reports/__snapshots__/generate-md-report.unit.test.ts.snap:109]: It's at the top of the expanded details, so I don't think it needs a heading. And I don't think a non-descriptive heading is useful. The heading should tell me what's in the table, and we can't do that generically in any meaningful way.

Comment (Tlacenka) [reply packages/utils/src/lib/reports/__snapshots__/generate-md-report.unit.test.ts.snap:109]: Shouldn't the title then be required if I need the title to know what's in the table?

**Review by Tlacenka — APPROVED** [review]
Nice improvement! 🎉

File: packages/utils/src/lib/reports/__snapshots__/generate-md-report.unit.test.ts.snap (hunk 1 above)
Comment (matejchalk) [reply packages/utils/src/lib/reports/__snapshots__/generate-md-report.unit.test.ts.snap:109]: Unfortunately that's not always possible. E.g. the tables from Lighthouse don't have titles, AFAIK.

But I would recommend plugin authors specify a title if possible.

**Pull request merged**